### ISSUE TITLE
``useless-else-on-loop`` should not be raised on `return`

### ISSUE THREAD

**Issue by tushar-deepsource** [PyCQA/pylint#5598]
### Bug description

Here's a typical example of W0120 being raised:

```python
def f():
    while True:
        if condition:
            pass
    else:
        pass
```

And this code **doesn't** raise W0120 (correctly):

```python
def f():
    while True:
        if condition:
            break
    else:
        pass
```

But, this code is pretty much identical in function and it **does** raise W0120 (bug):

```python
def f():
    while True:
        if condition:
            return
    else:
        pass
```

The check for W0120 should check for both `break` and `return` in the while block.

### Command used

```shell
pylint mytest.py
```

### Pylint output

```shell
************* Module mytest
mytest.py:5:4: W0120: Else clause on loop without a break statement (useless-else-on-loop)
```

### Expected behavior

Issue not raised

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6-dev0
Python 3.10.0 (default, Oct 13 2021, 06:45:00) [Clang 13.0.0 (clang-1300.0.29.3)]
```

**Comment by tushar-deepsource** [comment]
There's a few test cases in `W0120` that I don't quite understand:

```python
def test_return_for():
    """else + return is not acceptable."""
    for i in range(10):
        if i % 2:
            return i
    else:  # [useless-else-on-loop]
        print('math is broken')
    return None

def test_return_while():
    """else + return is not acceptable."""
    while True:
        return 1
    else:  # [useless-else-on-loop]
        print('math is broken')
    return None
```

Why should `W0120` be raised here?

**Comment by DanielNoord** [comment]
I'm not sure. It feels like they should indeed also not raise. Have you looked at the commit and PR that added them? Perhaps there is a discussion there?

**Comment by tusharsadhwani** [comment]
Removing those tests should make the PR ready for review, brb

**Comment by jacobtylerwalls** [comment]
> they actually discuss this and there seems to have been an internal discussion about this at the time

@DanielNoord That looks like the commit intending to do what @tushar-deepsource wants to do and permit `return` alongside `break`. Looks like they reversed themselves about a month later: 31e6b776a, giving the current behavior.

> Looking at the last line of the message I think there is a case to be made to remove this.

Meaning I think they expected they might go to the current behavior.

> Looking at the last line of the message I think there is a case to be made to remove this. I would vote in favour I think.

I guess I don't see a strong reason for the change; it just causes churn.
